Lib/http/server.py

```diff
@@ -1192,20 +1192,42 @@ def run_cgi(self):
                 nbytes = int(length)
             except (TypeError, ValueError):
                 nbytes = 0
+
+            data = None
+            rfile = None
+
+            if self.command.lower() == "post" and nbytes > 0:
```

Comment:
I don't have windows, so I didn't test even the existing code in Windows.  I believe that even the existing code is not going to work on windows due the condition nbytes > 0

nbytes is from `Content-Length` and I see that https://bugs.python.org/issue24764 Content-Length for multi-part form data was removed.

When I tried to add test, I could not exercise this.

* Before any change is done, a test case and testing under Windows is required for this Bugfix and Patch.